.github/workflows/build_and_push.yaml

```diff
@@ -0,0 +1,32 @@
+name: build_and_push
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build_and_push:
+    name: Build and push
+    runs-on: ubuntu-18.04
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Make
+      run: make
+
+    - name: Login to DockerHub
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKER_USER }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
```

Comment:
I don't think this will work because secrets are not exposed to PRs (as this would allow anyone to open a PR and get access to the secret...)

Reply:
Hmm yeah I wanted to avoid having two workflows/builds if only one was necessary, but you may be right that it won't be enough. I think we'll need to have `post_merge` publish too though if we want to have a tag tracking the master branch, right? I'll give it a go.